arch/arm64/core/smp.c

```diff
@@ -22,53 +23,103 @@
 #include <drivers/interrupt_controller/gic.h>
 #include <drivers/pm_cpu_ops.h>
 #include <sys/arch_interface.h>
+#include "boot.h"
 
 #define SGI_SCHED_IPI	0
 #define SGI_PTABLE_IPI	1
 
-volatile struct {
-	void *sp; /* Fixed at the first entry */
+struct boot_params {
+	uint64_t mpid;
+	char *sp;
 	arch_cpustart_t fn;
 	void *arg;
-} __aligned(L1_CACHE_BYTES) arm64_cpu_init[CONFIG_MP_NUM_CPUS];
+	int cpu_num;
```

Comment:
Don't need to reserve cpu_num anymoe because we use single boot_params for all secondary cores.